cylc/flow/scripts/validate_install_play.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""cylc validate-install-play [OPTIONS] ARGS
+
+
+- Should validate's arg be WORKFLOW|PATH as described or WORKFLOW|PATH|SOURCE?
+- cylc validate --output-filename probably isn't very useful to VIP and should
+  probably be forced unset.
+- cylc install -n (workflow name) clashed with cylc play -n (no-detach) -
+  proposal -n == --workflow-name because users are more likely to want that.
+  Consider logging a warning bc it's still ambiguous.
+- Possibe DANGER: cylc-rose options for cylc install (-S -O -D) could be
+  over-run at play by --set and --set-file, This can happen anyway,
+  but there's potential for it to be obscured by the VIP command, or even,
+  confusingly, fail validating after install which it passed before.
```

Comment:
This needs re-writing?

Reply:
Ay, it's a copy of the comment I made proposing how this ought to work!